qiita_pet/handlers/api_proxy/studies.py

```diff
@@ -436,6 +437,15 @@ def study_tags_patch_request(user_id, study_id,
             message = study.update_tags(User(user_id), req_value)
             return {'status': 'success',
                     'message': message}
+        elif attribute == 'specimen_id_column':
+            try:
+                study.specimen_id_column = req_value
```

Comment:
Wondering if this step (as it has to verify that the column is unique and set the SQL values) should be a job vs. a simple set ...

Reply:
This is only checking one column, so I don't think it will take that much time. The request should also be asynchronous from the UI so it shouldn't block any interaction. Maybe I'm overlooking something?